allensdk/brain_observatory/behavior/behavior_project_cache/external/behavior_project_metadata_writer.py

```diff
@@ -106,16 +101,43 @@ def _write_behavior_sessions(
                 include_trial_metrics=include_trial_metrics)
 
         # Add release files
-        behavior_sessions = behavior_sessions \
-            .merge(self._release_behavior_only_nwb,
-                   left_index=True,
-                   right_index=True,
-                   how='left')
-        if "file_id" in behavior_sessions.columns:
-            if behavior_sessions["file_id"].isnull().values.any():
-                msg = (f"{output_filename} field `file_id` contains missing "
-                       "values and pandas.to_csv() converts it to float")
-                warnings.warn(msg)
+        if self.args["behavior_nwb_dir"] is None:
+            behavior_sessions = behavior_sessions \
+                .merge(self._release_behavior_only_nwb,
+                       left_index=True,
+                       right_index=True,
+                       how='left')
+            if "file_id" in behavior_sessions.columns:
+                if behavior_sessions["file_id"].isnull().values.any():
+                    msg = (f"{output_filename} field `file_id` contains "
+                           "missing values and pandas.to_csv() converts it to "
+                           "float.")
+                    warnings.warn(msg)
+        else:
+            ophys_session_mask = behavior_sessions.ophys_session_id.isin(
+                self._released_ophys_session_ids
```

Comment:
please don't set the instance variable `self._released_ophys_session_ids`. it's confusing where it's set and makes it so that `write_ophys_experiments` has to be called before this?

why can't you just read list of ophys experiments from disk from the nwb files? Either that or just call `get_ophys_experiment_table` again to retrieve them or pass it in as an argument

Reply:
Changed to loading the experiment table.